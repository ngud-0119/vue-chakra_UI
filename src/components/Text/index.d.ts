import * as Vue from 'vue';

/**
 * The Text component is a composable text element.
 */
declare const Text: Vue.Component<{
  as?: String
}>;

export default Text;
