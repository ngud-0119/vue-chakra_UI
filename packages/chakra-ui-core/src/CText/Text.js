import CBox from '../Box'
import { forwardProps } from '../utils'
import { baseProps } from '../config/props'

export default {
  name: 'CText',
  inject: ['$theme', '$colorMode'],
  props: {
    as: {
      type: [String, Array],
      default: 'p'
    },
    ...baseProps
  },
  render (h) {
    return h(CBox, {
      props: {
        as: this.as,
        fontFamily: this.as === 'kbd' ? 'mono' : 'body',
        ...forwardProps(this.$props)
      }
    }, this.$slots.default)
  }
}
