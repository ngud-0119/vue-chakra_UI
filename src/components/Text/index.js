import { Box } from '../../lib/core'
import { forwardProps } from '../../lib/utils'
import { baseProps } from '../../lib/config/props'

export default {
  name: 'Text',
  inject: ['$theme', '$colorMode'],
  props: {
    as: {
      type: [String, Array],
      default: 'p'
    },
    ...baseProps
  },
  render (h) {
    return h(Box, {
      props: {
        as: this.as,
        ...forwardProps(this.$props)
      }
    }, this.$slots.default)
  }
}
