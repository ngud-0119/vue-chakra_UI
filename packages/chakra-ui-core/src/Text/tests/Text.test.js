import Text from '../'
import { render, defaultProviders } from '@/tests/test-utils'

const renderComponent = (props) => {
  const inlineAttrs = (props && props.inlineAttrs) || ''
  const base = {
    components: { KText: Text },
    provide: () => defaultProviders(),
    template: `<KText data-testid="text" ${inlineAttrs}>Text Works</KText>`,
    ...props
  }
  return render(base)
}

it('should render correctly', () => {
  const { asFragment } = renderComponent()

  expect(asFragment()).toMatchSnapshot()
})

it('should change the style', () => {
  const inlineAttrs = `d="flex"`
  const { asFragment, getByTestId } = renderComponent({ inlineAttrs })

  const text = getByTestId('text')

  expect(asFragment()).toMatchSnapshot()
  expect(text).toHaveStyle('display: flex')
})

it.each`
  as
  ${'u'}
  ${'abbr'}
  ${'cite'}
  ${'del'}
  ${'em'}
  ${'ins'}
  ${'kbd'}
  ${'mark'}
  ${'s'}
  ${'sub'}
  ${'sup'}}
`(
  'should display text type as $as',
  async ({ as }) => {
    const inlineAttrs = `as=${as}`
    const { asFragment } = renderComponent({ inlineAttrs })

    expect(asFragment()).toMatchSnapshot()
  }
)
