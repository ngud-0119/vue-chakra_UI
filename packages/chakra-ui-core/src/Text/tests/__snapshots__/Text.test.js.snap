// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change the style 1`] = `
<DocumentFragment>
  <p
    class="css-zlhv1"
    data-testid="text"
  >
    Text Works
  </p>
</DocumentFragment>
`;

exports[`should display text type as abbr 1`] = `
<DocumentFragment>
  <abbr
    class="css-fzcsno"
    data-testid="text"
  >
    Text Works
  </abbr>
</DocumentFragment>
`;

exports[`should display text type as cite 1`] = `
<DocumentFragment>
  <cite
    class="css-fzcsno"
    data-testid="text"
  >
    Text Works
  </cite>
</DocumentFragment>
`;

exports[`should display text type as del 1`] = `
<DocumentFragment>
  <del
    class="css-fzcsno"
    data-testid="text"
  >
    Text Works
  </del>
</DocumentFragment>
`;

exports[`should display text type as em 1`] = `
<DocumentFragment>
  <em
    class="css-fzcsno"
    data-testid="text"
  >
    Text Works
  </em>
</DocumentFragment>
`;

exports[`should display text type as ins 1`] = `
<DocumentFragment>
  <ins
    class="css-fzcsno"
    data-testid="text"
  >
    Text Works
  </ins>
</DocumentFragment>
`;

exports[`should display text type as kbd 1`] = `
<DocumentFragment>
  <kbd
    class="css-1r9kfq"
    data-testid="text"
  >
    Text Works
  </kbd>
</DocumentFragment>
`;

exports[`should display text type as mark 1`] = `
<DocumentFragment>
  <mark
    class="css-fzcsno"
    data-testid="text"
  >
    Text Works
  </mark>
</DocumentFragment>
`;

exports[`should display text type as s 1`] = `
<DocumentFragment>
  <s
    class="css-fzcsno"
    data-testid="text"
  >
    Text Works
  </s>
</DocumentFragment>
`;

exports[`should display text type as sub 1`] = `
<DocumentFragment>
  <sub
    class="css-fzcsno"
    data-testid="text"
  >
    Text Works
  </sub>
</DocumentFragment>
`;

exports[`should display text type as sup 1`] = `
<DocumentFragment>
  <sup
    class="css-fzcsno"
    data-testid="text"
  >
    Text Works
  </sup>
</DocumentFragment>
`;

exports[`should display text type as u 1`] = `
<DocumentFragment>
  <u
    class="css-fzcsno"
    data-testid="text"
  >
    Text Works
  </u>
</DocumentFragment>
`;

exports[`should render correctly 1`] = `
<DocumentFragment>
  <p
    class="css-fzcsno"
    data-testid="text"
  >
    Text Works
  </p>
</DocumentFragment>
`;
